test/functional/readconcern_tests.js

```diff
@@ -444,6 +444,55 @@ describe('ReadConcern', function() {
     }
   });
 
+  it('Should set majority readConcern option for aggregate command', {
+    metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2.0' } },
+
+    test: function(done) {
+      const listener = require('../..').instrument(function(err) {
+        expect(err).to.be.null;
+      });
+
+      // Contains all the apm events
```

Comment:
All of these tests are doing a _ton_ of the same boilerplate. I was wondering if we could incorporate two refactors, or at least consider them for the future:
  - convert all the tests to use some sort of table-driven approach to make it easier to very quickly add new tests in the future, e.g.
```
[
  {
    description: 'should set majority readConcern option for aggregate command',
    operation: coll => coll.aggregate([{ $match: {} }], { readConcern: { level: 'majority' } })
].forEach(testCase => {
  it(testCase.description, () => { ... });
});
```

  - provide some sort of helper function for "command monitoring" tests, something like:
```
it('should do something with command monitoring', commandMonitoringTest(done => {
  ... other setup ...
  return collection
    .aggregate([{ $match: {} }], { readConcern: { level: 'majority' } })
    .toArray(function(err) {
      const aggEvents = this.started.filter(evt => evt.name === 'aggregate');
      const toCheck = aggEvents[0];
      ... the rest ...
    });
}));
```
It would be responsible for instrumenting before the test, maintaining a list of started/failed/succeeded events for introspection, and uninstrumenting after the test. 

With these in place I think we could dramatically reduce the size of this file, improving readability and easing addition of new tests. I don't think we need to hold up this test, but if you're so inclined on a wednesday to play around with it I think it would be a great change 😄

Reply:
I think this is a good idea! I'm going to file a ticket for this and label it `neweng`. Either a new grad can do it or I can pick it up at a slow time :)